UI/TechTreeWnd.cpp

```diff
@@ -1583,20 +1604,108 @@ void TechTreeWnd::TechListBox::TechRow::Render() {
+            GG::X prop_width = total_width / 12;
+            DebugLogger() << "default config values for TechRow columns exceeded requested width: "
+                          << Value(default_cols_width_sum + desc_width) << "/" << Value(total_width)
+                          << ", setting to proportional widths";
+            graphic_width   = prop_width;
+            name_width      = prop_width * 3;
+            cost_width      = prop_width;
+            time_width      = prop_width;
+            category_width  = prop_width * 2;
+            desc_width      = prop_width * 4;
+        } else {
+            DebugLogger() << "config values for TechRow columns exceeded requested width: "
+                          << Value(cols_width_sum + desc_width) << "/" << Value(total_width)
+                          << ", set to default values";
+        }
+        GetOptionsDB().Set(TECHROW_COL_XML_PREFIX + ".graphic",     Value(graphic_width));
+        GetOptionsDB().Set(TECHROW_COL_XML_PREFIX + ".name",        Value(name_width));
+        GetOptionsDB().Set(TECHROW_COL_XML_PREFIX + ".cost",        Value(cost_width));
+        GetOptionsDB().Set(TECHROW_COL_XML_PREFIX + ".time",        Value(time_width));
+        GetOptionsDB().Set(TECHROW_COL_XML_PREFIX + ".category",    Value(category_width));
+
+        GetOptionsDB().Commit();
+
+    }
+
+    retval.push_back(graphic_width);
+    retval.push_back(name_width);
+    retval.push_back(cost_width);
+    retval.push_back(time_width);
+    retval.push_back(category_width);
+    retval.push_back(desc_width);
+
+    return retval;
+}
+
+std::vector<GG::Alignment> TechTreeWnd::TechListBox::TechRow::ColAlignments() {
+    std::vector<GG::Alignment> retval;
+
+    retval.push_back(GG::ALIGN_CENTER);  // graphic
+    retval.push_back(GG::ALIGN_LEFT);  // name
+    retval.push_back(GG::ALIGN_RIGHT);  // cost
+    retval.push_back(GG::ALIGN_RIGHT);  // time
+    retval.push_back(GG::ALIGN_LEFT);  // category
+    retval.push_back(GG::ALIGN_LEFT);  // description
+
+    return retval;
+}
+
+bool TechTreeWnd::TechListBox::TechRowCmp(const GG::ListBox::Row& lhs, const GG::ListBox::Row& rhs, std::size_t column) {
+    bool retval = false;
+    const std::string lhs_key = boost::trim_copy(lhs.SortKey(column));
+    const std::string rhs_key = boost::trim_copy(rhs.SortKey(column));
+
+    // When equal, sort by previous sorted column
+    if ((lhs_key == rhs_key) && (m_previous_sort_col != column)) {
+        retval = TechRowCmp(lhs, rhs, m_previous_sort_col);
+    } else if (column == 2 || column == 3) {
+        try {  // compare by int
+            retval = boost::lexical_cast<int>(lhs_key) < boost::lexical_cast<int>(rhs_key);
+        } catch (boost::bad_lexical_cast& e) {
+            DebugLogger() << "Unable to cast sort keys to int, lhs:" << lhs_key << " rhs:" << rhs_key << ". " << e.what();
+            retval = lhs_key < rhs_key;
+        }
+    } else {
+        retval = lhs_key < rhs_key;
```

Comment:
I've changed to a duplicated GetLocal(), I'm not quite sure how this is a better option when the locale is manually set to `en_US` however.
We could set these to `""` for the default system locale, until a proper link with the selected language is implemented.